core/input/input.cpp

```diff
@@ -697,16 +697,19 @@ void Input::_parse_input_event_impl(const Ref<InputEvent> &p_event, bool p_is_em
 	for (const KeyValue<StringName, InputMap::Action> &E : InputMap::get_singleton()->get_action_map()) {
 		if (InputMap::get_singleton()->event_is_action(p_event, E.key)) {
 			Action &action = action_state[E.key];
-			// If not echo and action pressed state has changed
-			if (!p_event->is_echo() && is_action_pressed(E.key, false) != p_event->is_action_pressed(E.key)) {
+			if (!p_event->is_echo()) {
 				if (p_event->is_action_pressed(E.key)) {
-					action.pressed = true;
-					action.pressed_physics_frame = Engine::get_singleton()->get_physics_frames();
-					action.pressed_process_frame = Engine::get_singleton()->get_process_frames();
+					action.pressed++;
+					if (action.pressed == 1) {
+						action.pressed_physics_frame = Engine::get_singleton()->get_physics_frames();
+						action.pressed_process_frame = Engine::get_singleton()->get_process_frames();
+					}
 				} else {
-					action.pressed = false;
-					action.released_physics_frame = Engine::get_singleton()->get_physics_frames();
-					action.released_process_frame = Engine::get_singleton()->get_process_frames();
+					if (action.pressed == 1) {
+						action.released_physics_frame = Engine::get_singleton()->get_physics_frames();
+						action.released_process_frame = Engine::get_singleton()->get_process_frames();
+					}
+					action.pressed = MAX(action.pressed - 1, 0);
```

Comment:
It solves a problem with modifier actions, like `ui_text_completion_query`. For some reason pressing them requires both keys (`Ctrl + Space` in case of the mentioned action), but releasing doesn't require the modifier to be pressed, which means they can be released without previously being pressed.